compiler/rustc_typeck/src/check/method/prelude2021.rs

```diff
@@ -258,7 +259,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             let self_ty_name = self
                 .sess()
                 .source_map()
-                .span_to_snippet(self_ty_span)
+                .span_to_snippet(self_ty_span.find_ancestor_inside(span).unwrap_or_default())
```

Comment:
Is `unwrap_or_default` what we want here?

Reply:
The `or_default` means that `span_to_snippet` gets a dummy span in those cases, which will result in the `self_ty.to_string()` case below.

I can rewrite it as a `.and_then` if you prefer.

Reply:
Updated it to use and_then.